flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -95,6 +99,21 @@
      */
     private int inFlightRequestsCount;
 
+    /**
+     * Tracks the cumulative size of all elements in {@code bufferedRequestEntries} to facilitate
+     * the criterion for flushing after {@code flushOnBufferSizeMB} is reached.
+     */
+    private double bufferedRequestEntriesTotalSizeMB;
```

Comment:
Why `double`? Are we expecting fractional values? Is it worth using `long` and bytes instead of MB for more flexibility?

Reply:
Yes I think so. I originally picked MB since I saw the elasticsearch sink use it. But looking at more sinks, it seems to be a personal choice. Bytes makes more sense for _us_ here.